pyclam/manifold.py

```diff
@@ -112,10 +109,12 @@ def __contains__(self, point: Data) -> bool:
 
     @property
     def parent(self) -> 'Cluster':
-        if self.depth > 0:
-            return self.manifold.ancestry(self)[-2]
-        else:
-            raise ValueError(f"root cluster has no parent")
+        if 'parent' not in self.cache:
```

Comment:
Should we just start spawning children with a `parent` attribute?

Reply:
done. see Cluster.partition